client/packages/android/app/src/main/java/org/openmsupply/client/NativeApi.java

```diff
@@ -349,30 +349,41 @@ public void onResolveFailed(NsdServiceInfo serviceInfo, int errorCode) {
 
     // Attempt to get a non-loopback address for the local server
     // and fallback to loopback if there is an error
-    private String getLocalAddress(NsdServiceInfo serviceInfo){
-        try {
-            for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements();) {
-                NetworkInterface intf = en.nextElement();
-                for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr.hasMoreElements();) {
-                    InetAddress inetAddress = enumIpAddr.nextElement();
-                    if (!inetAddress.isLoopbackAddress() && !inetAddress.isLinkLocalAddress() && inetAddress.isSiteLocalAddress()) {
-                        return inetAddress.getHostAddress();
+    private String getHostAddress(NsdServiceInfo serviceInfo, Boolean isLocal){
+        if (isLocal) {
+            try {
+                for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements(); ) {
+                    NetworkInterface intf = en.nextElement();
+                    for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr.hasMoreElements(); ) {
+                        InetAddress inetAddress = enumIpAddr.nextElement();
+                        if (!inetAddress.isLoopbackAddress() && !inetAddress.isLinkLocalAddress() && inetAddress.isSiteLocalAddress()) {
+                            return inetAddress.getHostAddress();
+                        }
                     }
                 }
+            } catch (Exception ex) {
+                Log.e(OM_SUPPLY, ex.toString());
             }
-        } catch (SocketException ex) {
+        }
+        try {
+            return serviceInfo.getHost().getHostAddress();
+        } catch (Exception ex) {
```

Comment:
nit: maybe nicer to use finally and one exception ?
```java
    private String getHostAddress(NsdServiceInfo serviceInfo, Boolean isLocal) {
        try {
            if (isLocal) {
                for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en
                        .hasMoreElements();) {
                    NetworkInterface intf = en.nextElement();
                    for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr.hasMoreElements();) {
                        InetAddress inetAddress = enumIpAddr.nextElement();
                        if (!inetAddress.isLoopbackAddress() && !inetAddress.isLinkLocalAddress()
                                && inetAddress.isSiteLocalAddress()) {
                            return inetAddress.getHostAddress();
                        }
                    }
                }
            }

            return serviceInfo.getHost().getHostAddress();
            
        } catch (Exception ex) {
            Log.e(OM_SUPPLY, ex.toString());
        } finally {
            // with no network available the getHostAddress() will
            // throw an Exception - in this case default to local loopback
            // as no other network is reachable!
            return "127.0.0.1";
        }
    }
```

Reply:
yep, nicer. thanks!